codice/ddf
# DDF-5439 Decouple UI and Service Facet Whitelist

#### What does this PR do?
Breaks Facet Whitelist into two whitelists; one at the service level and one for the UI
#### Who is reviewing it? 
@andrewzimmer 
@hayleynorton 
@zta6 

#### Ask 2 committers to review/merge the PR and tag them here.
@djblue 
@bdeining 

#### How should this be tested?
1. Build / Install / Ingest a document or two
2. In the admin console, add some text attribute names to `Catalog UI Search Attribute Suggestion List`
3. Refresh Intrigue. Select some of those attributes from query advanced. Verify that the input is NOT populated with a dropdown on those attributes
4. In the admin console, add some text attribute names to `Facet Attribute Whitelist` (with some overlap from step 2)
5. Refresh Intrigue. Select some of those attributes from query advanced. Verify the ones that were added to both configs populate the input with a dropdown. Verify the others do not.
6. In postman query the `/internal/cql` endpoint and verify that all attributes from step 4 are facetable. An example request looks like:
```
{
    "cql": "anyText ILIKE '%'",
    "facets": ["title"]
}
``` 
7. Any other spot testing appropriate.

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #5439 

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
